pkg/monitor/monitor.go

```diff
@@ -91,16 +94,20 @@ func (mon *monitor) Run(ctx context.Context) error {
 		err = mon.listBuckets(ctx)
 		if err != nil {
 			mon.baseLog.Error(err)
+		} else {
+			mon.lastBucketlist.Store(time.Now())
 		}
 
 		<-t.C
 	}
 }
 
-// checkReady checks the ready status of the frontend to make it consistent
+// checkReady checks the ready status of the monitor to make it consistent
 // across the /healthz/ready endpoint and emitted metrics.   We wait for 2
 // minutes before indicating health.  This ensures that there will be a gap in
 // our health metric if we crash or restart.
 func (mon *monitor) checkReady() bool {
-	return time.Now().Sub(mon.startTime) > 2*time.Minute
+	return (time.Now().Sub(mon.lastBucketlist.Load().(time.Time)) < time.Minute) && // did we list buckets successfully recently?
+		(time.Now().Sub(mon.lastChangefeed.Load().(time.Time)) < time.Minute) && // did we process the change feed recently?
```

Comment:
if either of these atomic.Values does not yet contain a time.Time when this is called, then there will be a panic.